### PR TITLE
List permissions fix No. 3 in a series

### PR DESCRIPTION
**Pull request by ckolos**
So maybe there are two ways to create a list in shavar and maybe we use
one not the other. Let's be sure and set object acl's either way.

I blame someone else for this not being fixed the first time.

### CODE REVIEW

File: publish2cloud.py [hunk 1]
```diff
@@ -205,7 +206,8 @@ def publish_to_s3(config, section, chunknum):
     for key_name in (chunk_key, key):
         k = boto.s3.key.Key(bucket)
         k.key = key_name
-        k.set_contents_from_filename(output_filename, policy='bucket-owner-full-control')
+        k.set_contents_from_filename(output_filename)
+        k.set_acl('bucket-owner-full-control')
```
Comment (skim1102) [inline publish2cloud.py]:
```suggestion
        k.set_acl('public-read')
```
According to the documentation [here](http://boto.cloudhackers.com/en/latest/s3_tut.html#setting-getting-the-access-control-list-for-buckets-and-keys), if we want to have the files be publicly readable the policy should be `public-read`.

File: publish2cloud.py [hunk 2]
```diff
@@ -154,6 +154,7 @@ def new_data_to_publish_to_s3(config, section, new):
             key.set_contents_from_string('a:1:32:32\n' + 32 * '1')
         current = tempfile.TemporaryFile()
         key.get_contents_to_file(current)
+        key.set_acl('bucket-owner-full-control')
```
Comment (skim1102) [inline publish2cloud.py]:
```suggestion
        key.set_acl('public-read')
```

**Comment by ckolos-moz-admin** [comment]
After chatting it out with our security folk and AWS support, no one could state that it was a terrible idea to have the cloudfront object access id in public, but no one would also commit to it not being a terrible idea.

I've switched the backend job for shavar list generation to pass in the necessary credentials and tested list generation both from my local host as well as from jenkins. All works as expected. I:

- Deleted the digest256 file
- deleted the chunk file
- re-ran list-generation to recreate the file
- verified that the permissions on the re-created files now included the cloudfront distro's access
- verified that I could pull down the resources from cloudfront using the form: <CDN Url>/<key>

I then re-set the permissions on all existing files to allow cloudfront read and tested *all* files currently in stage.

I think this works.

**Comment by ckolos** [comment]
After chatting it out with our security folk and AWS support, no one could state that it was a terrible idea to have the cloudfront object access id in public, but no one would also commit to it not being a terrible idea either.

I've switched the backend job for shavar list generation to pass in the necessary credentials and tested list generation both from my local host as well as from jenkins. All works as expected. I:

- Deleted the digest256 file
- deleted the chunk file
- re-ran list-generation to recreate the file
- verified that the permissions on the re-created files now included the cloudfront distro's access
- verified that I could pull down the resources from cloudfront using the form: <CDN Url>/<key>

I then re-set the permissions on all existing files to allow cloudfront read and tested *all* files currently in stage.

I think this works.

**Pull request merged**